src/java.base/share/native/libjli/java.c

```diff
@@ -534,6 +545,260 @@ invokeInstanceMainWithoutArgs(JNIEnv *env, jclass mainClass) {
+            if (NULL != equals) {
+                if ('\0' == *(equals + 1)) {
+                    return NULL;
+                }
+                int length = strlen(equals + 1);
+                char *value = (char *)JLI_MemAlloc(length + 1);
+                if (NULL == value) {
+                    JLI_ReportErrorMessage(
+                        "Failed to allocate memory for the value of the command line option %s.",
+                        optionName);
+                    *error = MEMORY_ALLOCATION_ERROR;
+                    return NULL;
+                }
+                snprintf(value, length + 1, "%s", equals + 1);
+                value[length] = '\0';
+                return value;
+            }
+            return NULL;
+        }
+    }
+    *error = OPTION_NOT_FOUND;
+    return NULL;
+}
+
+static void freeCommandLineOptionValue(char *value) {
+    if (value) {
+        JLI_MemFree(value);
+        value = NULL;
+    }
+}
+
+static jboolean isCommandLineOptionFoundWithError(int error) {
+    return OPTION_NOT_FOUND != error && error;
+}
+
+static char *getCheckpointDir(int argc, char **argv, int *error) {
+    const char *checkpointDirPropertyName = "-XX:CRaCRestoreFrom";
+    char *checkpointDirPropertyValue = getCommandLineOptionValue(checkpointDirPropertyName, argc, argv, error);
+    if (!*error) {
+        if (checkpointDirPropertyValue) {
+            return checkpointDirPropertyValue;
+        } else {
+            JLI_ReportErrorMessage("The value of the command line option %s was not found.", checkpointDirPropertyName);
+            *error = OPTION_VALUE_NOT_FOUND;
+            return NULL;
+        }
+    }
+    return NULL;
+}
+
+static char *getDefaultLogLevel(int *error) {
+    *error = 0;
+    char *defaultLogLevel = strdup("-v2");
+    if (NULL == defaultLogLevel) {
+        JLI_ReportErrorMessage("Failed to allocate memory for the CRIU default log level.");
+        *error = MEMORY_ALLOCATION_ERROR;
+    }
+    return defaultLogLevel;
+}
+
+static char *getLogLevel(int argc, char **argv, int *error) {
+    const char *logLevelPropertyName = "-Dopenj9.internal.criu.logLevel";
+    char *logLevelPropertyValue = getCommandLineOptionValue(logLevelPropertyName, argc, argv, error);
+    if (!isCommandLineOptionFoundWithError(*error)) {
+        if (logLevelPropertyValue) {
+            for (const char *c = logLevelPropertyValue; *c; c++) {
+                if (!isdigit(*c)) {
+                    JLI_ReportErrorMessage(
+                        "The value %s of the command line option %s is not valid.",
+                        logLevelPropertyValue,
+                        logLevelPropertyName);
+                    freeCommandLineOptionValue(logLevelPropertyValue);
+                    *error = INVALID_OPTION_VALUE;
+                    return NULL;
+                }
+            }
+            int logLevelValue = atoi(logLevelPropertyValue);
+            if (0 <= logLevelValue && 4 >= logLevelValue) {
```

Comment:
```suggestion
            if ((logLevelValue >= 0) && (logLevelValue <= 4)) {
```
Put the `rvalue` on the left only for comparing two values for equality or inequality.

Reply:
I suggest it reads more clearly expressed in the form `(min <= x) && (x <= max)`:
```c
            if ((0 <= logLevelValue) && (logLevelValue <= 4)) {
```

Reply:
How can I add the appropriate guideline to the coding standard? The following line of code seems to align closely with the [omr coding standard](https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md): 

```
if ((logLevelValue >= 0) && (logLevelValue <= 4)) {
```

Reply:
@JasonFengJ9 @keithc-ca how can I know the answer to the following question: how can I add the appropriate guideline to the coding standard?

Reply:
@tajila @JasonFengJ9 @keithc-ca how can I know the answer to the following question: how can I add the appropriate guideline to the coding standard?

Reply:
You could open a pull request https://github.com/eclipse/omr with proposed changes to https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md.

Reply:
I would like to provide a [pull request](https://github.com/eclipse/omr/pull/7363) with the proposed changes to https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md.